src/error.rs

```diff
@@ -1,11 +1,19 @@
+
 use defines::AfError;
 use std::error::Error;
+use std::sync::Mutex;
 
 pub type ErrorCallback = Fn(AfError);
 
 pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
```

Comment:
I have tried passing ErrorCallback to Mutex, but that didn't compile for me. Therefore, i created the mutex around an integer.

Yes, i see it now, this is wrong. Won't help protect the data.